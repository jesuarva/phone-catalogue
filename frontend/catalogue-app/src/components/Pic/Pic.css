.pic img {
  min-height: 50vh;
}