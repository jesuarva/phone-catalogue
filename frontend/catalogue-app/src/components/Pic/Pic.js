import PropTypes from 'prop-types';
import React from 'react';
import './Pic.css';

const Pic = ({ img, altAttribute }) => {
  return (
    <div className="pic">
      <figure className="figure">
        <img
          src={img}
          className="figure-img img-fluid rounded"
          alt={altAttribute}
        />
      </figure>
    </div>
  );
};

Pic.propTypes = {
  img: PropTypes.string.isRequired,
  altAttribute: PropTypes.string.isRequired,
};

export default Pic;
