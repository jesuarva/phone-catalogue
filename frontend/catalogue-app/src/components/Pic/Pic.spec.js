import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import MAXMOVIL_LOGO from '../../media/Masmovillogo.svg.png';
import Pic from './Pic';

configure({ adapter: new Adapter() });

describe('Pic component', () => {
  const picComponent = <Pic img={MAXMOVIL_LOGO} altAttribute="Maxmovil logo" />;

  it('should renders correctly', () => {
    const tree = renderer.create(picComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
