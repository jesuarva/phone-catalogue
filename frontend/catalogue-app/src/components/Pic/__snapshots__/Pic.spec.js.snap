// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pic component should renders correctly 1`] = `
<div
  className="pic"
>
  <figure
    className="figure"
  >
    <img
      alt="Maxmovil logo"
      className="figure-img img-fluid rounded"
      src="Masmovillogo.svg.png"
    />
  </figure>
</div>
`;
